packages/ruleset/src/functions/use-date-based-format.js

```diff
@@ -0,0 +1,305 @@
+      `${ruleId}: logical schema path is : ${logicalPathForLogger(logicalPath)}`
+    );
+
+    // Use a composition-aware utility to gather any examples relevant to this
+    // schema, including those defined on applicator schemas in oneOf, etc.
+    const schemaExamples = getExamplesForSchema(schema);
+    logger.debug(`${ruleId}: ${schemaExamples.length} examples found`);
+
+    // Examples have already been stored on the parent - as we go through the
+    // schemas, check for the presence of an example value for the current
+    // property within the parent's example.
+    const parentalExamples =
+      // If the logical path is empty, there are no parents, but there may be
+      // indirect examples to add in the ":" branch.
+      logicalPath.length > 0
+        ? // Look at the example values for the logical parent schema, if any.
+          // For successive, nested properties, this will end up propagating
+          // examples through the recursive descent so that example values
+          // separated from a property by multiple degrees of nesting will
+          // still be preserved.
+          examples[logicalPath.slice(0, -1).join('.')]
+            .map(e => {
+              // Check the parental example values for
+              // the presence of the current property.
+              const prop = logicalPath.at(-1);
+
+              // Check for sentinel indicating an array.
+              if (prop === '[]' && Array.isArray(e)) {
+                return e[0];
+              }
+
+              // Check for sentinel indicating a dictionary.
+              if (prop === '*' && isObject(e)) {
+                return Object.values(e)[0];
+              }
+
+              // Standard model path.
+              return e[prop];
+            })
+
+            // We are not guaranteed to find a value - filter
+            // out any values that are not defined.
+            .filter(e => e !== undefined)
+        : // Add indirect examples to the map - note that they will necessarily
+          // be indexed with the empty string key, like primary schemas.
+          indirectExamples;
+
+    logger.debug(
+      `${ruleId}: ${parentalExamples.length} examples found in logical parent`
+    );
+
+    // Index the examples with the stringified logical path.
+    // Note that the unconditional assignment is intentional - there may be
+    // existing entries for this same logical path (e.g. for the same
+    // nested property within a different oneOf sibling) but we want to
+    // override them, always. Otherwise, the behavior would depend on the
+    // order the schemas are checked in (the logic may look at more examples
+    // for one instance of a property than another, arbitrarily) and we don't
+    // make a guarantee that order will be stable in `validateNestedSchemas`.
+    examples[logicalPath.join('.')] = [...schemaExamples, ...parentalExamples];
+
+    return performValidation(schema, path, logicalPath, examples);
```

Comment:
No biggie, but there's technically no need to pass `logicalPath` and the full `examples` map in here, right? Because we're pre-collecting the parental examples, `performValidation()` is only ever going to use `examples[logicalPath.join('.')]`.

Sorry, this popped into my head over the weekend and then I forgot about it.

Reply:
Oh good catch - that's right! I'll refactor.